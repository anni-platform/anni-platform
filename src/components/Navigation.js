import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router';
import { getAuthUrl } from 'adapters';
import { login, logoutSession } from 'adapters';
import { addAuthToken, logout } from 'actions';

class Navigation extends Component {
  componentDidMount() {
    const { dispatch, isAuthenticated } = this.props;
    if (!isAuthenticated) {
      login().then(token => {
        if (!token) {
          this.props.router.push("/");
          return;
        }
        dispatch(addAuthToken(token));
      }, (err) => console.log(err) );
    }
  }
  logout() {
    logoutSession();
    this.props.dispatch(logout());
    this.props.router.push("/");
  }
  render() {
    const login = (
      <div>
        <a href={getAuthUrl()}>Login To Dropbox</a>
      </div>
    );
    const loggedInNav = (
      <nav className='navigation'>
        <Link to="/dashboard">Dashboard</Link>
        <Link to="/patterns">Patterns</Link>
        <button className="buttonLink" onClick={this.logout.bind(this)}>Logout</button>
      </nav>
    )
    return (
      <div className="Navigation">
        {(this.props.auth.isAuthenticated ? loggedInNav : login)}
      </div>
    );
  }
}

const mapStateToProps = ({ auth }) => ({
  auth
});

export default connect(mapStateToProps)(withRouter(Navigation));
