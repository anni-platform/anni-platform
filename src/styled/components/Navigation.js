import styled from 'styled-components';
import { FLINT, PAPER } from './Variables';

export const NavBar = styled.div`
  background: ${PAPER};
  border-bottom: 1px solid ${FLINT};
  bottom: ${props => (props.bottom ? '0' : '')};
  align-items: center;
  display: flex;
  height: 72px;
  left: 0;
  min-width: 100vw;
  overflow: ${props => props.scroll && 'scroll'};
  position: fixed;
  top: ${props => (props.bottom ? '' : '0')};
  width: 100%;
  z-index: 3;
`;

export const NavItem = styled.div`
  align-items: center;
  border-left: 1px solid ${FLINT};
  display: flex;
  flex-grow: ${props => props.fit && '1'};
  justify-content: center;
  height: 100%;
  margin-left: ${props => props.right && 'auto'};
  padding: 0;
`;

export const NavItemGroup = styled.div`
  align-items: center;
  display: flex;
  height: 100%;
  margin-left: ${props => props.right && 'auto'};
`;
