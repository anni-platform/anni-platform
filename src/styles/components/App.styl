.Loading {
    padding: 1rem;
}