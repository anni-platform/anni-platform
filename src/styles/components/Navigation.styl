.Navigation
  align-items center
  border-bottom 1px solid $concrete
  display flex
  justify-content space-between
  padding $small $medium
  position relative

  +above($mqMedium)
    padding $medium $large

  a
    font-family $headlineFont
    letter-spacing 1px
    text-transform uppercase
    padding 0 $medium

  a.icon
    margin 0 $small
    transition all 400ms $expo

    +above($mqMedium)
      margin 0 $medium 0 $tiny

    &:hover
      fill $copper
      transform: scale(1.1)

  a.link
    display inline-block
    font-family $headlineFont
    letter-spacing 1px
    margin 0 $medium
    padding $small 0
    position relative
    text-transform uppercase

    +above($mqMedium)
      margin 0 $medium
      padding $medium 0

    &:link, &:visited
      text-decoration none

    &:after
      background $copper
      bottom 7px
      content ''
      height 3px
      left 0
      position absolute
      transition all 400ms $expo
      width 0

    &:hover
      &:after
        width 100%

  a.selected
    &:after
      background $copper
      bottom 7px
      content ''
      height 3px
      left 0
      position absolute
      transition all 800ms $expo
      width 100%

  .profile
    @extend button.circle
    background $copper
    color $paper
    height 40px
    width 40px

.NavigationItems
  align-items center
  display flex
