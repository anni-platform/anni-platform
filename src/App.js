import React, { Component } from 'react';
import { createStore, compose } from 'redux';
import Navigation from 'components/Navigation';
import { Provider } from 'react-redux';
import reducer from 'reducers';
import { getAccessTokenFromUrl, storeSessionToken, getAuthUrl } from 'adapters';
import { saveState } from './utils/localStorage';
import StaticJSONFileDatabase from 'utils/fileStorage';
import { Loader, Wrapper } from 'styled';
import GlobalStyles from 'styled/components/Base';

let enhancer = null;
let DevTools = null;
if (process.env.NODE_ENV === 'development') {
  DevTools = require('DevTools').default;
  enhancer = compose(DevTools.instrument());
}

const test = {
  a: 1,
  b: 2,
};

class App extends Component {
  constructor() {
    super();
    this.state = { store: null };
  }
  componentDidMount() {
    const token = getAccessTokenFromUrl();
    window.loginDropbox = function() {
      window.location.href = getAuthUrl();
    };
    if (token) {
      storeSessionToken(token);
    }
    StaticJSONFileDatabase.hydrateStoreFromFileDatabase().then(data => {
      let store;
      if (process.env.NODE_ENV === 'development') {
        store = createStore(reducer, data, enhancer);
      } else {
        store = createStore(reducer, data);
      }
      store.subscribe(() => {
        saveState(store.getState());
      });
      this.setState({ store });
    });
  }
  render() {
    GlobalStyles();

    const store = this.state.store;
    return store ? (
      <Provider store={store}>
        <Wrapper>
          <Navigation />
          {this.props.children}
          {DevTools ? <DevTools /> : null}
        </Wrapper>
      </Provider>
    ) : (
      <Loader fullPage />
    );
  }
}

export default App;
