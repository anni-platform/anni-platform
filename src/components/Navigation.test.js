import React from 'react';
import App from 'App';
import Navigation from './Navigation';
import renderer from 'react-test-renderer';

describe('Navigation Component', () => {
  it('renders without crashing', () => {
    const component = renderer.create(
      <App>
        <Navigation />
      </App>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
