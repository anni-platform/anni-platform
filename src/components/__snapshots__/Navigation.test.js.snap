exports[`Navigation Component renders without crashing 1`] = `
<div
  className="Loading">
  Loading...
</div>
`;
